doc/faq/howto_faq.rst

```diff
@@ -37,6 +37,38 @@ If you only want to use the `pandas` converter for `numpy.datetime64` values ::
 
 
 
+.. _howto-figure-empty:
+
+Check whether a figure is empty
+-------------------------------
+Empty can actually mean different things. Does the figure contain any artists?
+Does a figure with an empty `~.axes.Axes` still count as empty? Is the figure
+empty if there it would be rendered pure white (there may be artists present,
+but they could be outside the drawing area or transparent).
+
+For the purpose here, we define empty as: "The figure does not contain any
+artists except it's background patch." The exception for the background is
+necessary, because by default every figure contains a `.Rectangle` as it's
+background patch. This definition could be checked via::
+
+    def is_empty(figure):
+        """
+        Return whether the figure contains no Artists (other than the default
+        background patch).
+        """
+        contained_artists = figure.findobj(include_self=False)
+        return (
+            len(contained_artists) < 1 or
+            len(contained_artists) == 1 and contained_artists[0] is figure.patch)
```

Comment:
I guess you could simplify this to `len(contained_artists) < 2`, unless you would know of a way to get rid of the background patch.

Reply:
Indeed, `fig.patch.remove()` does not work.